src/linker/implementation.py

```diff
@@ -37,7 +37,6 @@ def run(
         session: Optional["drmaa.Session"],
         runner: Callable,
         step_id: str,
```

Comment:
I didn't think about this at all, since it didn't have any bearing on the feature at hand, but it looks like it is used by the logger. It is determined by pipeline.py:

`        number_of_steps = len(self.implementations)
        number_of_steps_digit_length = len(str(number_of_steps))
        for idx, implementation in enumerate(self.implementations):
            step_number = str(idx + 1).zfill(number_of_steps_digit_length)
            previous_step_number = str(idx).zfill(number_of_steps_digit_length)
            step_id = f"{step_number}_{implementation.step_name}"`

It's one of those things that `could` be determined by the step but is difficult to have defined by the step in the current structure of the code.